src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.EventDispatch.cs

```diff
@@ -34,6 +34,14 @@ internal Task DispatchSubmitEventAsync(string? handlerName, out bool isBadReques
 
         if (!_namedSubmitEventsByScopeQualifiedName.TryGetValue(handlerName, out var locationsForName) || locationsForName.Count == 0)
         {
+            if (isReExecuted)
+            {
+                // If we are re-executing, we do not expect to find a new form on the page we re-execute to.
+                // This does not mean that the request is bad, we want the re-execution to succeed.
+                isBadRequest = false;
+                return Task.CompletedTask;
+            }
+
```

Comment:
This feels like it's at the wrong place.

The moment we are re-executing I don't think we want to even try to run any of the form dispatching logic. I think we want to move this logic to around line 109 in `src/Components/Endpoints/src/RazorComponentEndpointInvoker.cs`